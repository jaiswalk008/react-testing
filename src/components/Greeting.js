const Greeting = () =>{
    return (
        <div>
            <h1>Hello World!!</h1>
            <p>Its good to see you</p>
            <strong>Testing...</strong>
        </div>
    )
}
export default Greeting;