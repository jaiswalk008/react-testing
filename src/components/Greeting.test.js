import Greeting from "./Greeting";
import { render , screen} from "@testing-library/react";

describe('Greeting component' ,() =>{
    test('renders Hello World as a text' , () =>{
        //Arrange - set up the test data, test conditions, and test environment
        render(<Greeting/>);
    
    
        //Act - execute the function
    
        //assert- check the result
        //exact - casing wont matter and also includes substrings
        const helloWorldElement = screen.getByText('Hello World',{exact:false });
  
        expect(helloWorldElement).toBeInTheDocument();
       
    });
    test('renders greeting paragraph', () => {
        render(<Greeting />);
        const paragraphElement = screen.getByText('Its good to see you');
        expect(paragraphElement).toBeInTheDocument();
    });
    test('contains a heading element', () => {
        render(<Greeting />);
        const headingElement = screen.getByRole('heading');
        expect(headingElement).toBeInTheDocument();
    });
    test('renders with the correct text content', () => {
        render(<Greeting />);
        const headingElement = screen.getByRole('heading');
        expect(headingElement.textContent).toBe('Hello World!!');
        
    });
    test('renders testing heading', () => {
        render(<Greeting />);
        const testingElement = screen.getByText('Testing...');
        expect(testingElement).toBeInTheDocument();
    });
    
})
