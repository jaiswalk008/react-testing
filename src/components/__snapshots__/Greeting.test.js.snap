// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Hello World!!
    </h1>
    <p>
      It's good to see you
    </p>
  </div>
</DocumentFragment>
`;
